ocaml/xapi-idl/lib/posix_channel.ml

```diff
@@ -4,62 +4,7 @@ exception End_of_file
 
 exception Channel_setup_failed
 
-module CBuf = struct
-  (** A circular buffer constructed from a string *)
-  type t = {
-      mutable buffer: bytes
-    ; mutable len: int  (** bytes of valid data in [buffer] *)
-    ; mutable start: int  (** index of first valid byte in [buffer] *)
-    ; mutable r_closed: bool  (** true if no more data can be read due to EOF *)
-    ; mutable w_closed: bool
-          (** true if no more data can be written due to EOF *)
-  }
-
-  let empty length =
-    {
-      buffer= Bytes.create length
-    ; len= 0
-    ; start= 0
-    ; r_closed= false
-    ; w_closed= false
-    }
-
-  let drop (x : t) n =
-    if n > x.len then failwith (Printf.sprintf "drop %d > %d" n x.len) ;
-    x.start <- (x.start + n) mod Bytes.length x.buffer ;
-    x.len <- x.len - n
-
-  let should_read (x : t) =
-    (not x.r_closed) && x.len < Bytes.length x.buffer - 1
-
-  let should_write (x : t) = (not x.w_closed) && x.len > 0
-
-  let end_of_reads (x : t) = x.r_closed && x.len = 0
-
-  let end_of_writes (x : t) = x.w_closed
-
-  let write (x : t) fd =
-    (* Offset of the character after the substring *)
-    let next = min (Bytes.length x.buffer) (x.start + x.len) in
-    let len = next - x.start in
-    let written =
-      try Unix.single_write fd x.buffer x.start len
-      with _e ->
-        x.w_closed <- true ;
-        len
-    in
-    drop x written
-
-  let read (x : t) fd =
-    (* Offset of the next empty character *)
-    let next = (x.start + x.len) mod Bytes.length x.buffer in
-    let len =
-      min (Bytes.length x.buffer - next) (Bytes.length x.buffer - x.len)
-    in
-    let read = Unix.read fd x.buffer next len in
-    if read = 0 then x.r_closed <- true ;
-    x.len <- x.len + read
-end
+module CBuf = Xapi_stdext_unix.Unixext.CBuf
```

Comment:
This will be obsoleted by this commit which deletes this file entirely:

3c5c28e78aabe22d5d46dfe5c5a507e80ad6ae4b CP-47536: Drop posix_channel and channel_helper: unused and a mix of Unix/Lwt